ecommerce/extensions/payment/views/lumsxpay.py

```diff
@@ -44,16 +44,11 @@ def extract_items_from_basket(self, basket):
             for l in basket.all_lines()
         ]
 
-    def get_existing_basket(self, request):
-        courses = [i['id'] for i in self.extract_items_from_basket(request.basket)]
-        course_ids = []
+    def get_existing_basket_challan(self, request):
+        basket = request.basket
+        product = basket.lines.first().product
```

Comment:
are we sure about the availability of the `basket` object, it can't be None?

Reply:
yes `first()` wont break because it will only be called once a object is in basket

Reply:
Actually the comment was targeted for first-line `basket = request.basket`

Reply:
yes it wont break too